lib/lists.nix

```
@@ -1766,6 +1766,9 @@ rec {
   /**
     Remove duplicate elements from the `list`. O(n^2) complexity.
 
+    :::{.note}
+    If the list contains only strings complexity can be reduced to O(n) by using [`lib.lists.uniqueStrings`](#function-library-lib.lists.uniqueStrings) instead.
```

```suggestion
    If the list only contains strings, the complexity can be reduced to O(n log n) by using [`lib.lists.uniqueStrings`](#function-library-lib.lists.uniqueStrings) instead.
```

Nix performs a sort internally on symbol ids, which is $$n\ \log\ n$$. Probably a $$\log\ n$$ that pales in comparison to interpretation overhead etc.